src/main/java/com/photolocos/enterprise/service/PhotoService.java

```diff
@@ -14,20 +17,26 @@
 
     @Autowired
     IPhotoDAO photoDAO;
+    @Autowired
+    ILocationDAO locationDAO;
 
 
     @Override
-    public PhotoDTO fetchByLocation(String location) {
-
-        return null;
+    @Cacheable(value = "fetchByArea", key = "#area")
+    public Set<PhotoDTO> fetchByArea(String area) throws Exception {
+        LocationDTO location = new LocationDTO();
+        location.setArea(area);
+        return photoDAO.fetchByLocation(location);
     }
 
     @Override
-    public Set<PhotoDTO> fetchByTag(String tag) {
-        return null;
+    @Cacheable(value = "photosByTag", key = "#tag")
+    public Set<PhotoDTO> fetchByTag(String tag) throws Exception {
+        return photoDAO.fetchByTag(new String[]{tag});
     }
 
     @Override
+    @CacheEvict(value = {"fetchByArea", "photosByTag", "photosByStateAndCity", "allPhotos"}, allEntries = true)
```

Comment:
Added caching of photo sets returned from the service layer. All cache is evicted when a new photo is saved.